src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -0,0 +1,291 @@
+
+    // initialize a weighted tree node
+    node = std::make_shared<weighted_tree_node_t> (parent_bank,
+                                                   bank_name,
+                                                   false,
+                                                   std::stoll (bank_shares),
+                                                   0);
+
+    // if there is no parent bank, then the root pointer points to the root bank
+    if (!parent_bank) {
+        root_ptr = node;
+    } else {
+        parent_bank->add_child (node);
+    }
+
+    rc = sqlite3_bind_text (b_select_sub_banks_stmt,
+                            1,
+                            bank_name.c_str (),
+                            -1,
+                            NULL);
+    if (rc != SQLITE_OK) {
+        fprintf (stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        return nullptr;
+    }
+    rc = sqlite3_step (b_select_sub_banks_stmt);
+
+    // vector of tuples to store usernames and their respective shares
+    std::vector<std::tuple<std::string, std::string, std::string>> users;
+
+    // vector of strings to hold sub banks
+    std::vector<std::string> banks;
+
+    // we've reached a bank with no sub banks, so print
+    // out all associations under this sub bank
+    if (rc != SQLITE_ROW) {
+        rc = sqlite3_bind_text (b_select_associations_stmt,
+                                1,
+                                bank_name.c_str (),
+                                -1,
+                                NULL);
+        if (rc != SQLITE_OK) {
+            fprintf(stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+            sqlite3_close (DB);
+
+            return nullptr;
+        }
+
+        // execute SQL statement
+        rc = sqlite3_step (b_select_associations_stmt);
+        while (rc == SQLITE_ROW) {
+            std::string user = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 0));
+            std::string single_user_shares = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 1));
+            std::string job_usage = reinterpret_cast<char const *> (
+                sqlite3_column_text (b_select_associations_stmt, 3));
+            // place the user, their allocated shares, and
+            // their job_usage into vector
+            users.emplace_back (user, single_user_shares, job_usage);
+            rc = sqlite3_step (b_select_associations_stmt);
+        }
+        // iterate through associations and add them as children
+        // of node
+        for (unsigned int i = 0; i < users.size (); i++) {
+            std::string username = std::get<0> (users[i]);
+            std::string user_shares = std::get<1> (users[i]);
+            std::string user_usage = std::get<2> (users[i]);
+            user_node = std::make_shared<weighted_tree_node_t> (
```

Comment:
I would move this tree node construction up into the while loop above, and then you can remove the `users` vector and this `for` loop.  That will shorten the code a bit and reduce the amount of looping.